doc/user/content/sql/patterns/window-functions.md

```diff
@@ -1,8 +1,126 @@
 ---
-title: "Window functions"
-description: "Use lateral joins as a streaming-efficient stand-in for many window functions."
-draft: true
+title: "Window functions (OVER clause)"
+description: "Avoid window functions, e.g., use lateral joins as a streaming-efficient stand-in for many window functions."
 menu:
   main:
     parent: 'sql-patterns'
 ---
+
+Materialize currently does not handle window functions (`OVER` clause) efficiently for large partition sizes: when an input record is added/removed/changed, the system recomputes the results for the entire partition that the changed record belongs to. We are planning to make window functions more efficient in Materialize, but for the time being, this page shows some workarounds for avoiding the use of window functions.
+
+Note that this page **does _not_ refer to temporal windows**, i.e., when a window belongs to a time interval. Examples of temporal windows are when we select only the last hour of an infinite stream of records, or when we group input records by which hour they occurred in. Such queries are efficiently supported in Materialize by using [temporal filters and temporal groupings](/sql/patterns/temporal-filters).
+
+By "window functions", this page means functions such as `LAG`, `LEAD`, `ROW_NUMBER`, `FIRST_VALUE`. You can access these functions with the `OVER` clause, in which you use the `PARTITION BY` clause to group input records into partitions by a key, and use the `ORDER BY` clause to order records inside each partition. Window functions compute results based on the position of records inside these partitions. For example, `x - LAG(x)` computes the differences of attribute `x` between each pair of _consecutive_ rows in each partition.
```

Comment:
```suggestion
```

Reply:
You mean we should delete this paragraph?